packages/cosmwasm/src/cw3cosmwasmclient.ts

```diff
@@ -0,0 +1,197 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { BroadcastMode, GasLimits, GasPrice, OfflineSigner } from "@cosmjs/launchpad";
+
+import { CosmWasmFeeTable, ExecuteResult, SigningCosmWasmClient } from "./signingcosmwasmclient";
+
+export type Expiration =
+  | {
+      readonly at_height: number;
+    }
+  | {
+      readonly at_time: number;
+    };
+
+export enum Vote {
+  Yes = "yes",
+  No = "no",
+  Abstain = "abstain",
+  Veto = "veto",
+}
+
+export interface ThresholdResult {
+  readonly absolute_count: {
+    readonly weight_needed: number;
+    readonly total_weight: number;
+  };
+}
+
+export interface ProposalResult {
+  readonly id: number;
+  readonly title: string;
+  readonly description: string;
+  readonly msgs: ReadonlyArray<Record<string, unknown>>;
+  readonly expires: Expiration;
+  readonly status: string;
+}
+
+export interface ProposalsResult {
+  readonly proposals: readonly ProposalResult[];
+}
+
+export interface VoteResult {
+  readonly vote: Vote;
+}
+
+export interface VotesResult {
+  readonly votes: ReadonlyArray<{ readonly vote: Vote; readonly voter: string; readonly weight: number }>;
+}
+
+export interface VoterResult {
+  readonly addr: string;
+  readonly weight: number;
+}
+
+export interface VotersResult {
+  readonly voters: readonly VoterResult[];
+}
+
+interface StartBeforeNumberPaginationOptions {
+  readonly startBefore?: number;
+  readonly limit?: number;
+}
+
+interface StartAfterNumberPaginationOptions {
+  readonly startAfter?: number;
+  readonly limit?: number;
+}
+
+interface StartAfterStringPaginationOptions {
+  readonly startAfter?: string;
+  readonly limit?: number;
+}
+
+export class Cw3CosmWasmClient extends SigningCosmWasmClient {
```

Comment:
It would be good to easily query the balance of the contract. When using this client, we assume we are acting on behalf of the multisig and our senderAddress is used to sign our txs, but otherwise unimportant.

However, `getBalances()` will be returning info about `senderAddress`, not `cw3ContractAddress`, right? I think we should use that contract address as the default account to query. (And/or expose it somehow from the client, the senderAddress is `public readonly`, correct?)